detox/src/devices/drivers/android/emulator/AVDValidator.js

```diff
@@ -1,28 +1,54 @@
 const _ = require('lodash');
 const path = require('path');
-const AVDsResolver = require('./AVDsResolver');
 const environment = require('../../../../utils/environment');
+const logger = require('../../../../utils/logger').child({ __filename });
+
+const REQUIRED_EMULATOR_MAJOR = 29;
 
 class AVDValidator {
-  constructor(emulatorExec) {
-    this._avdsResolver = new AVDsResolver(emulatorExec);
+  constructor(avdsResolver, emulatorVersionResolver) {
+    this._avdsResolver = avdsResolver;
+    this._emulatorVersionResolver = emulatorVersionResolver;
   }
 
   async validate(avdName) {
     const avds = await this._avdsResolver.resolve(avdName);
+    this._assertAVDs(avds);
+    await this._assertAVDMatch(avds, avdName);
+    await this._validateEmulatorVer();
+  }
+
+  _assertAVDs(avds) {
     if (!avds) {
       const avdmanagerPath = path.join(environment.getAndroidSDKPath(), 'tools', 'bin', 'avdmanager');
```

Comment:
Should belong to `utils/environment.js`, i.e. `environment.getAVDManagerPath()`.